### PR TITLE
Improvements to multi-envrionments based on feedback

### PR DESCRIPTION
**Pull request by kaustavghosh06**
* Added `amplify env add` command as a proxy instead of amplify init command (users complained of it being confusing)

* Changed the previous `amplify env add` command to `amplify env import` command to bring in an external cloudformation stack 

* Added further checks on the params needed for the `amplify env import` command.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-interactions/provider-utils/awscloudformation/cloudformation-templates/lex-cloudformation-template.json.ejs [hunk 1]
```diff
@@ -269,6 +269,7 @@
                                       {
                                         "Ref": "AWS::Region"
                                       },
+                                      ":",
```
Comment (yuth) [inline packages/amplify-category-interactions/provider-utils/awscloudformation/cloudformation-templates/lex-cloudformation-template.json.ejs]: Is this relevant to push command?

File: packages/amplify-cli/src/commands/env/add.js [hunk 2]
```diff
@@ -1,60 +1,10 @@
-const fs = require('fs-extra');
-const path = require('path');
+const { run } = require('../init');
```
Comment (yuth) [inline packages/amplify-cli/src/commands/env/add.js]: Do we ask the question if its an existing env when this command is run

File: packages/amplify-cli/src/commands/env/import.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        awsInfo = JSON.parse(context.parameters.options.awsInfo);
+      } catch (e) {
+        context.print.error('You must pass in the AWS credential info in an object format for intializating your environment using the --awsInfo flag');
+        process.exit(1);
+      }
+    }
+
+    let envFound = false;
+    const allEnvs = context.amplify.getEnvDetails();
+    Object.keys(allEnvs).forEach((env) => {
```
Comment (yuth) [inline packages/amplify-cli/src/commands/env/import.js]: Nit: instead of using `forEach` you could use [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) to do the search

File: packages/amplify-cli/src/commands/env/import.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      if (env === envName) {
+        envFound = true;
+      }
+    });
+
+    const addNewEnvConfig = () => {
```
Comment (yuth) [inline packages/amplify-cli/src/commands/env/import.js]: This can be its own file so that it can be tested with unit test. Longer term we should move all the configuration related thing to a single module

File: packages/amplify-cli/src/commands/env/import.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      }
+
+      envAwsInfo[envName] = awsInfo;
+      jsonString = JSON.stringify(envAwsInfo, null, 4);
+      fs.writeFileSync(configInfoFilePath, jsonString, 'utf8');
+
+      context.print.success('Successfully added environment from your project');
+    };
+
+    if (envFound) {
```
Comment (yuth) [inline packages/amplify-cli/src/commands/env/import.js]: Nit: we probably can simply the whole thing with 
```javascript
let shouldAdd = context.paramters.options.yes;
if (envFound && !context.paramters.options.yes) {
shouldAdd = await context.amplify.confirmPrompt.run('We found an environment with the same name. Do you want to overwrite existing enviornment config?')
}
if(shouldAdd) {
   addNewEnvConfig(); 
}

```

File: packages/amplify-cli/src/lib/init-steps/s0-analyzeProject.js [hunk 6]
```diff
@@ -123,7 +123,7 @@ async function getEnvName(context) {
   const isEnvNameValid = (inputEnvName) => {
     let valid = true;
 
-    if (inputEnvName.length > 10 || inputEnvName.length < 2 || /[^a-zA-Z]/g.test(inputEnvName)) {
+    if (inputEnvName.length > 10 || inputEnvName.length < 2 || /[^a-z]/g.test(inputEnvName)) {
```
Comment (yuth) [inline packages/amplify-cli/src/lib/init-steps/s0-analyzeProject.js]: why only lower case support?

File: packages/amplify-cli/src/commands/env/add.js (hunk 2 above)
Comment (kaustavghosh06) [reply packages/amplify-cli/src/commands/env/add.js]: Yes. We do.

File: packages/amplify-cli/src/lib/init-steps/s0-analyzeProject.js (hunk 6 above)
Comment (kaustavghosh06) [reply packages/amplify-cli/src/lib/init-steps/s0-analyzeProject.js]: because resource names like S3 don't support upper case. This is an artifact of this bug/issue -> https://github.com/aws-amplify/amplify-cli/issues/373#issuecomment-458689994

Comment (yuth) [reply packages/amplify-cli/src/lib/init-steps/s0-analyzeProject.js]: OK, Can we have a separate commit for this. So its easier to track

Comment (kaustavghosh06) [reply packages/amplify-cli/src/lib/init-steps/s0-analyzeProject.js]: Yes, there is a separate commit for it -> https://github.com/aws-amplify/amplify-cli/pull/803/commits/6c581481c083c3af6f4c7a2eab489755b92ec4b4

File: packages/amplify-category-interactions/provider-utils/awscloudformation/cloudformation-templates/lex-cloudformation-template.json.ejs (hunk 1 above)
Comment (kaustavghosh06) [reply packages/amplify-category-interactions/provider-utils/awscloudformation/cloudformation-templates/lex-cloudformation-template.json.ejs]: Rebased.
